scylla/src/network/connection.rs

```diff
@@ -1848,7 +1848,23 @@ pub(super) async fn open_connection(
     };
 
     // If this is ScyllaDB that we connected to, we received sharding information.
-    let shard_info = ShardInfo::try_from(&supported.options).ok();
+    let shard_info = match ShardInfo::try_from(&supported.options) {
+        Ok(info) => Some(info),
+        Err(ShardingError::NoShardInfo) => {
+            tracing::info!(
+                "[{}] No sharding information received. Proceeding with no sharding info.",
+                addr
+            );
+            None
+        }
+        Err(e) => {
+            tracing::warn!(
+                "[{}] Error while parsing sharding information: {}. Proceeding with no sharding info.",
+                addr, e
+            );
+            None
+        }
```

Comment:
:thinking: I think this deserves `ERROR` level. After all, this _**is**_ an error.